File: mod/1.20.x/fabric/src/main/java/net/wiredtomato/waygl/fabric/service/FabricPlatformService.java
package net.wiredtomato.waygl.fabric.service;

import com.google.auto.service.AutoService;
import net.fabricmc.loader.api.FabricLoader;
import net.wiredtomato.waygl.core.service.PlatformService;

import java.nio.file.Path;

@AutoService(PlatformService.class)
public class FabricPlatformService implements PlatformService {
    @Override
    public String platform() {
        return "net/wiredtomato/waygl/fabric";
    }

    @Override
    public Boolean isModLoaded(String modId) {
        return FabricLoader.getInstance().isModLoaded(modId);
    }

    @Override
    public Boolean isDevelopmentEnvironment() {
        return FabricLoader.getInstance().isDevelopmentEnvironment();
    }

    @Override
    public String getMinecraftVersionString() {
        return FabricLoader.getInstance().getModContainer("minecraft").orElseThrow().getMetadata().getVersion().getFriendlyString();
    }

    @Override
    public Path getConfigDirectory() {
        return FabricLoader.getInstance().getConfigDir();
    }
}

File: mod/fabric/src/main/kotlin/net/wiredtomato/waygl/service/FabricPlatformService.kt
package net.wiredtomato.waygl.service

import deplatformed.ServiceImpl
import net.fabricmc.loader.api.FabricLoader
import java.nio.file.Path

@ServiceImpl([PlatformService::class])
class FabricPlatformService : PlatformService {
    override fun platform(): String = "fabric"
    override fun isModLoaded(modId: String): Boolean = FabricLoader.getInstance().isModLoaded(modId)
    override fun isDevelopmentEnvironment(): Boolean = FabricLoader.getInstance().isDevelopmentEnvironment
    override fun getMinecraftVersion(): String = FabricLoader.getInstance().getModContainer("minecraft").orElseThrow().metadata.version.friendlyString
    override fun getConfigDir(): Path = FabricLoader.getInstance().configDir
}